sdk/tables/azure-data-tables/samples/async_samples/sample_update_upsert_merge_entities_async.py

```diff
@@ -111,10 +111,8 @@ async def list_all_entities(self):
                 await table.create_entity(entity=entity1)
                 # [START list_entities]
                 # Query the entities in the table
-                i = 0
-                async for entity in table.list_entities():
+                async for i, entity in table.list_entities():
```

Comment:
`list_entities` doesn't return two outputs by default - I think this would need to be:
`async for i, entity in enumerate(table.list_entities())`
However I've never tried using `enumerate` with an async iterator before...

Reply:
It failed with `TypeError: 'AsyncItemPaged' object is not iterable`, I'll define `i` outside.

Btw, the `list_entities()` is a **sync** function, but it use **async** keyword in most cases when iterating _listXXX()_ result, I'll ship a separate PR to address that.

Reply:
That's correct - `list_entities` is a sync function. The async behaviour is only in the iterator.
But you're right - I think `i` needs to be defined outside the loop. That seems to be a gap in Python that I hope they address some day...

Reply:
Ah, got it! Will add the `async` back.